certbot/plugins/selection.py

```diff
@@ -39,6 +39,35 @@ def pick_authenticator(
     return pick_plugin(
         config, default, plugins, question, (interfaces.IAuthenticator,))
 
+def get_unprepared_installer(config, plugins):
+    """
+    Get an unprepared interfaces.IInstaller object.
+
+    :param certbot.interfaces.IConfig config: Configuration
+    :param certbot.plugins.disco.PluginsRegistry plugins:
+        All plugins registered as entry points.
+
+    :returns: Unprepared installer plugin.
+    :rtype: IPlugin
+    """
+
+    _, req_inst = cli_plugin_requests(config)
+    if not req_inst:
+        raise errors.MissingCommandlineFlag(
+            "Missing command line flags. For non-interactive execution, "
+            "you will need to specify an installer plugin on the command line. "
+            "You can do this by running Certbot with '--installer PLUGIN_NAME ")
+    _installers = plugins.filter(lambda p_ep: p_ep.name == req_inst)
+    if _installers:
+        inst = list(_installers.values())[0]
```

Comment:
In the extremely unlikely case there's two+ installers with the same name, we're just arbitrarily picking one. I think the most common case of this would a name collision with third party plugins. Since picking the wrong plugin could cause problems down the line, how about we fail fast here and check that `len(_installers) == 1` in the `if` statement above? If it's not 1, we'll error out below.

Reply:
Nice catch. This is now implemented.